kratos/processes/apply_ray_casting_interface_recognition_process.h

```diff
@@ -44,8 +44,8 @@ class KRATOS_API(KRATOS_CORE) ApplyRayCastingInterfaceRecognitionProcess : publi
     /// Pointer definition of ApplyRayCastingInterfaceRecognitionProcess
     KRATOS_CLASS_POINTER_DEFINITION(ApplyRayCastingInterfaceRecognitionProcess);
 
-    KRATOS_REGISTRY_ADD_TEMPLATE_PROTOTYPE("Processes.KratosMultiphysics", Process, ApplyRayCastingInterfaceRecognitionProcess, TDim)
-    KRATOS_REGISTRY_ADD_TEMPLATE_PROTOTYPE("Processes.All", Process, ApplyRayCastingInterfaceRecognitionProcess, TDim)
+    KRATOS_REGISTRY_ADD_TEMPLATE_PROTOTYPE("Processes.KratosMultiphysics", Process, ApplyRayCastingInterfaceRecognitionProcess<TDim>, int[TDim])
```

Comment:
Just one thing before merge, how would be the usage for non-type template parameters?

Reply:
Exactly as the line you have marked `type[value]`, so int[TDim]` translates to `template<int TDim>`